Makefile

```diff
@@ -1,52 +1,191 @@
-DEFAULT_GOARCH := $(shell go env GOARCH)
-BUILD_HASH = $(shell git rev-parse HEAD)
-LDFLAGS += -X "github.com/mattermost/rtcd/service.buildHash=$(BUILD_HASH)"
-
-## Check go mod files consistency
-.PHONY: gomod-check
-gomod-check:
-	@echo Checking go mod files consistency
-	go mod tidy -v && git --no-pager diff --exit-code go.mod go.sum || (echo "Please run \"go mod tidy\" and commit the changes in go.mod and go.sum." && exit 1)
-
-.PHONY: check-style
-check-style: golangci-lint gomod-check
-	@echo Checking for style guide compliance
-
-.PHONY: golangci-lint
-golangci-lint: ## Run golangci-lint on codebase
-# https://stackoverflow.com/a/677212/1027058 (check if a command exists or not)
-	@if ! [ -x "$$(command -v golangci-lint)" ]; then \
-		echo "golangci-lint is not installed. Please see https://github.com/golangci/golangci-lint#install for installation instructions."; \
-		exit 1; \
-	fi; \
-
-	@echo Running golangci-lint
-	golangci-lint run ./...
 
-.PHONY: test
-test:
-	go test -v -mod=readonly -failfast -race ./...
+# Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+# See LICENSE for license information.
+
+# ====================================================================================
+# Variables
+# App Go Code location
+CONFIG_APP_CODE        += ./cmd/rtcd
+# Dockerfile's location
+CONFIG_DOCKERFILE      += ./build/Dockerfile
+# Docker options to inherit for all docker run commands
+CONFIG_DOCKEROPTS      += --rm -i -u $$(id -u):$$(id -g) --platform "linux/amd64"
+# Registry to upload images
+CONFIG_IMAGE_REGISTRY  += my.registy.com
+CONFIG_DOCKER_REPO     += mattermost
+# Temporary folder to output compiled binaries artifacts
+CONFIG_BIN_DIR         := dist
+## Docker Images Variables
+IMAGE_GOLINT           += "golangci/golangci-lint:v1.45.2@sha256:e84b639c061c8888be91939c78dae9b1525359954e405ab0d9868a46861bd21b"
```

Comment:
Using append(`+=`) mode for the variables is intended? For instance IMAGE_GOLINT make target will fail if user want to use his own image. Is it okey to use conditional assignment (`?=`) ?

```
ustafa@mmmmm:/srv/github/rtcd$ IMAGE_GOLINT=golangci/golangci-lint:v1.45.2 make go-lint
10:22:01 [ .. ] App linting...
docker: Error response from daemon: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "golangci/golangci-lint:v1.45.2@sha256:e84b639c061c8888be91939c78dae9b1525359954e405ab0d9868a46861bd21b": stat golangci/golangci-lint:v1.45.2@sha256:e84b639c061c8888be91939c78dae9b1525359954e405ab0d9868a46861bd21b: no such file or directory: unknown.
10:22:02 [FAIL]
make: *** [Makefile:173: go-lint] Error 1
```

Reply:
Nice. Haven't have thought this usage overwriting the configuration on runtime. 

Yes using conditional assignment would work in this case. But is this really needed? 
Is there a use case that the user of makefile would need to overwrite makefile variables, instead of directly editing the makefile and commiting the required changes? 

@streamer45  What is your take on this? Is there such a use case?

Reply:
There are definitely use cases to override variables without going through the trouble of editing the file but it really depends on what the variable is about. Changing the image here may not be one of them.

Reply:
Ok, Let's resolve this for now, and let @streamer45  provide some feedback after using it for sometime. 
What do you say @pfltdv ?

Reply:
Yes lets wait for developers response, will resolve this thread.